src/components/Courses/CoursesList.tsx

```diff
@@ -1,17 +1,41 @@
 import { Routes, Route } from 'react-router-dom';
-import { useState, useMemo } from 'react';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { all as getCourses } from 'store/courses/coursesSlice';
+import { allCourses } from 'store/courses/selectors';
+import { all as getAuthors } from 'store/authors/authorsSlice';
+import { allAuthors } from 'store/authors/selectors';
+
+import api from 'utils/api';
 
 import CourseInfo from './components/CourseInfo/CourseInfo';
 import Courses from './Courses';
 import ProtectedRoute from 'common/ProtectedRoute/ProtectedRoute';
 import CreateCourse from 'components/CreateCourse/CreateCourse';
 
-import { mockedCoursesList, mockedAuthorsList } from 'constants/constants';
 import Path from 'constants/Path';
 
 function CoursesList() {
-	const [courses, setCourses] = useState(mockedCoursesList);
-	const [authors, setAuthors] = useState(mockedAuthorsList);
+	const dispatch = useDispatch();
+	const courses = useSelector(allCourses);
+	const authors = useSelector(allAuthors);
+
+	useEffect(() => {
+		fetchCoursesInfo();
```

Comment:
_Это не нужно фиксить, просто инфа для тебя:_
Если задавалась вопросом, почему у тебя два раза моунтится компонент, тоесть запросы вызываются дважды - это из-за <StrictMode> в index.tsx

https://www.google.com/search?q=with+StrictMode+double+mount+react&rlz=1C1GCEA_enUA980UA980&oq=with+StrictMode+double+mount+react&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIHCAEQABiiBDIHCAIQABiiBDIHCAMQABiiBDIHCAQQABiiBNIBCTE1NDk2ajBqMagCALACAA&sourceid=chrome&ie=UTF-8
https://stackoverflow.com/questions/61254372/my-react-component-is-rendering-twice-because-of-strict-mode
https://stackoverflow.com/questions/72489140/react-18-strict-mode-causing-component-to-render-twice